src/main/java/de/tum/in/www1/artemis/domain/enumeration/Language.java

```diff
@@ -4,5 +4,34 @@
  * The Language enumeration.
  */
 public enum Language {
-    ENGLISH, GERMAN
+
+    ENGLISH, GERMAN;
+
+    /**
+     * Returns the language of for a given language short name.
+     *
+     * @param languageShortName the short name of the language
+     * @return the language that matches the given short name
+     */
+    public static Language fromLanguageShortName(String languageShortName) {
+        return switch (languageShortName) {
+            case "de" -> GERMAN;
+            case "en" -> ENGLISH;
+            default -> throw new IllegalArgumentException("Language not supported");
+        };
+    }
+
+    public static boolean isValidShortName(String language) {
+        return switch (language.toLowerCase()) {
+            case "de", "en" -> true;
+            default -> false;
+        };
+    }
+
+    public String getShortName() {
+        return switch (this) {
+            case GERMAN -> "de";
+            case ENGLISH -> "en";
+        };
+    }
```

Comment:
Consider changing to that the language names are only hardcoded once:
```suggestion
    ENGLISH("en"),
    GERMAN("de");

    private final String shortName;

    Language(String shortName) {
        this.shortName = shortName;
    }

    public String getShortName() {
        return this.shortName;
    }

    public static Language fromLanguageShortName(String languageShortName) {
        for (Language language : Language.values()) {
            if (language.getShortName().equals(languageShortName)) {
                return language;
            }
        }
        throw new IllegalArgumentException("Language not supported");
    }

    public static boolean isValidShortName(String languageShortName) {
        for (Language language : Language.values()) {
            if (language.getShortName().equals(languageShortName)) {
                return true;
            }
        }
        return false;
    }
```